storm-core/src/jvm/backtype/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -0,0 +1,296 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.blobstore;
+
+import backtype.storm.Config;
+import backtype.storm.generated.SettableBlobMeta;
+import backtype.storm.generated.AuthorizationException;
+import backtype.storm.generated.KeyAlreadyExistsException;
+import backtype.storm.generated.KeyNotFoundException;
+import backtype.storm.generated.ReadableBlobMeta;
+
+import backtype.storm.nimbus.NimbusInfo;
+import backtype.storm.utils.Utils;
+import org.apache.curator.framework.CuratorFramework;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.Subject;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;;
+
+import static backtype.storm.blobstore.BlobStoreAclHandler.ADMIN;
+import static backtype.storm.blobstore.BlobStoreAclHandler.READ;
+import static backtype.storm.blobstore.BlobStoreAclHandler.WRITE;
+
+/**
+ * Provides a local file system backed blob store implementation for Nimbus.
+ */
+public class LocalFsBlobStore extends BlobStore {
+  public static final Logger LOG = LoggerFactory.getLogger(LocalFsBlobStore.class);
+  private static final String DATA_PREFIX = "data_";
+  private static final String META_PREFIX = "meta_";
+  protected BlobStoreAclHandler _aclHandler;
+  private final String BLOBSTORE_SUBTREE = "/blobstore/";
+  private NimbusInfo nimbusInfo;
+  private FileBlobStoreImpl fbs;
+  private Map conf;
+
+  @Override
+  public void prepare(Map conf, String overrideBase, NimbusInfo nimbusInfo) {
+    this.conf = conf;
+    this.nimbusInfo = nimbusInfo;
+     if (overrideBase == null) {
+       overrideBase = (String)conf.get(Config.BLOBSTORE_DIR);
+      if (overrideBase == null) {
+        overrideBase = (String) conf.get(Config.STORM_LOCAL_DIR);
+      }
+    }
+    File baseDir = new File(overrideBase, BASE_BLOBS_DIR_NAME);
+    try {
+      fbs = new FileBlobStoreImpl(baseDir, conf);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    _aclHandler = new BlobStoreAclHandler(conf);
+  }
+
+  @Override
+  public AtomicOutputStream createBlob(String key, SettableBlobMeta meta, Subject who) throws AuthorizationException, KeyAlreadyExistsException {
+    LOG.debug("Creating Blob for key {}", key);
+    validateKey(key);
+    _aclHandler.normalizeSettableBlobMeta(key, meta, who, READ | WRITE | ADMIN);
```

Comment:
I've seen READ | WRITE | ADMIN occur a few times, should we define a constant for this?